instaclustr/resource_vpc_peering.go

```diff
@@ -112,16 +92,16 @@ func resourceVpcPeeringCreate(d *schema.ResourceData, meta interface{}) error {
 	if err != nil {
 		return fmt.Errorf("[Error] Error creating VPC peering request: %s", err)
 	}
-
-	id, err := client.CreateVpcPeering(cdcID, jsonStr)
+	var id string
+	id, err = client.CreateVpcPeering(cdcID, jsonStr)
 	if err != nil {
 		return fmt.Errorf("[Error] Error creating cluster: %s", err)
 	}
 	d.SetId(id)
 	d.Set("vpc_peering_id", id)
 	d.Set("cdc_id", cdcID)
-
-	vpcPeering, err := client.ReadVpcPeering(cdcID, id)
+	var vpcPeering *VPCPeering
+	vpcPeering, err = client.ReadVpcPeering(cdcID, id)
```

Comment:
**[Question]** Why do we need to explicitly state that `vpcPeering` is a pointer to a VPCPeering instance? I like that it's more descriptive, just wondering why.

Reply:
For some reason it was giving me invalid type error till I stated that explicitly

Reply:
If you have two different read funcs for AWS and GCP and you are re-using the variable then that's probably why. Pointers are unconvertible once type fixated.